server/portal/apps/intromessages/views.py

```diff
@@ -0,0 +1,43 @@
+"""
+.. :module: apps.intromessages.views
+   :synopsis: Views to handle read/unread status of IntroMessages
+"""
+
+import logging
+from portal.views.base import BaseApiView
+from django.http import JsonResponse
+from portal.apps.intromessages.models import IntroMessages
+from django.contrib.auth.decorators import login_required
+from django.utils.decorators import method_decorator
+import json
+
+
+logger = logging.getLogger(__name__)
+
+
+@method_decorator(login_required, name='dispatch')
+class IntroMessagesView(BaseApiView):
+    # Get all of the IntroMessages that have been read (unread = False)
+    def get(self, request, *args, **kwargs):
+        messages_array = IntroMessages.objects.filter(user=request.user).values('component', 'unread').values()
+        messages = [{'component': message['component'], 'unread': message['unread']} for message in messages_array]
+        return JsonResponse({'response': messages})
+
+    def put(self, request, *args):
+        body = json.loads(request.body)
+        # get all of the IntroMessages stored in the database
+        db_messages = IntroMessages.objects.filter(user=request.user).values()
+        for component_name, component_value in body.items():
+            found = False
+            for db_message in db_messages:          # Check to see if it's already stored in database
+                if db_message['component'] == component_name:
+                    found = True                    # Yes, does its unread status need to be updated in the database?
+                    if db_message['unread'] != component_value['unread']:
+                        db_message['unread'] = component_value['unread']
+                        db_message_object = IntroMessages(user=request.user, id=db_message['id'], component=component_name, unread=component_value['unread'])
+                        db_message_object.save()
+            if not found:                           # No, so we need to store in database
+                new_message_object = IntroMessages(user=request.user, component=component_name, unread=component_value['unread'])
+                new_message_object.save()
```

Comment:
I think this could be refactored a bit via the following:
1. Search for the `IntroMessages` object by also including the `component`
2. Adding a `def mark_read` function to an `IntroMessages` object, like for Notifications
3. Let the `mark_read` handle the logic to save the updated object

Reply:
Done.  Refactored code following suggested changes.